### DISCUSSION TITLE
ESP32 to UNO + CNC Shield

### DISCUSSION THREAD

**Discussion by Fazmat7** [HomeSpan/HomeSpan#800] [Q&A]
Hi, I'm currently trying to automate my blinds and I am using and esp32 for the HomeSpan side, and an uno r4 for the motorisation side. I'm hoping to use the r4 as an active passthrough for the esp chip so that, for example, if i wanted to close the blinds, the esp sends a signal through one of the pins to the uno, the uno recognises this and performs a set of instructions that closes the blinds through the CNC shield. I was wondering how to connect the esp to the uno since the CNC shield uses or covers all of the uno's input pins, and also if this arrangement would work or if the esp board requires a more direct connection to the motors.

Any help is appreciated.

**Comment by aalcaponix** [comment]
This was described for an Arduino (R3) with a CNC shield to connect a Bluetooth module. ESP32 connects the same way. 
Check if this helps..
https://forum.arduino.cc/t/solved-send-grbl-to-cnc-shield-through-esp32/1043923/6
